mountpoint-s3-client/src/object_client.rs

```diff
@@ -336,6 +336,11 @@ pub struct ObjectInfo {
     /// https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html#API_HeadObject_Examples
     pub storage_class: Option<String>,
 
+    /// Objects with GLACIER or DEEP_ARCHIVE storage classes are only acessable after restoration
+    /// This field is is set to true when \1 object has either GLACIER or DEEP_ARCHIVE storage class \2 restoration is fully
+    /// completed (i.e. head_object response contains "x-amz-restore" header with ongoing-request="false").
+    pub restored: bool,
+
```

Comment:
While we're here, maybe we should do the slightly richer thing, define an enum like this:
```rust
pub enum RestoreStatus {
    InProgress,
    Restored { expiry: SystemTime },
}
```
and then make this field `pub restored: Option<RestoreStatus>`.

I don't think we actually need to _use_ the expiry time for anything, though it's nice to have.

Reply:
+1 that we should make it `Option` and using enum would be great.

Reply:
its a good thing to do, it lets us analyse expiry for 2 purposes:
1. object is not readable if `expiry` is less than `now()`;
2. inode ttl is better to be less than `expiry`;

While \1 is simple, wasn't able to tackle \2 as inode expiry is measured in `Instant` which is not comparable with "human" dates. I gave up since I assume inode ttl's precision is in fractions of a second, which means that its unlikely that we'll cache "valid" restoration status after the object's been put back to archive by S3. Let me know, if you have any other thoughts on that, pls.